src/components/common/PageSearch/PageSearch.tsx

```diff
@@ -0,0 +1,79 @@
+import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward'
+import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward'
+import SearchIcon from '@mui/icons-material/Search'
+import { Box, IconButton, InputAdornment, TextField } from '@mui/material'
+import { observer } from 'mobx-react-lite'
+import React, { ChangeEvent, useEffect, useState } from 'react'
+
+import { useSharedStore } from '../../SharedStoreProvider'
+
+import { useStyles } from './PageSearch.styles'
+
+const PageSearchInput = () => {
+    const [inputValue, setInputValue] = useState('')
```

Comment:
why not to put that value into the store instead of state here

Reply:
I feel like there is no need of moving these small states and updates to the store. This approach makes the component independent and reusable. I know we want to use the MobX store as much as possible, thus the refactoring you made. But I don't think we have to dismiss local states entirely. 

The first time i wrote the component, it was exactly with the store state and reactions, and then i changed it like this because it is not necessary for the state to live in the store. 

This is definitely subjective, but it has no impact on the codes readability, quality or performance

Reply:
putting state into store creates separations of concerns, letting component be just the presentation layer while stores holds state and logic, it do makes code more readable and maintainable

Reply:
ok, i've changed this